packages/victory-core/src/victory-state/use-scale.ts

```diff
@@ -0,0 +1,67 @@
+import * as React from "react";
+import {
+  scaleLinear,
+  scaleTime,
+  scaleLog,
+  scaleSqrt
+} from "victory-vendor/d3-scale";
+import { AxisType, D3ScaleFn, ScalePropType } from "../types/prop-types";
+import * as Collection from "../victory-util/collection";
+import { getValueForAxis, isFunction } from "../victory-util/type-helpers";
+import { VictoryProviderProps } from "./types";
+import { useAxisData } from "./use-axis-data";
+
+type Scale = ScalePropType | D3ScaleFn;
+
+type ScaleProps = Pick<VictoryProviderProps, "data" | "scale">;
+
+const DEFAULT_SCALE = scaleLinear;
+
+function isD3Scale(scale?: unknown): scale is D3ScaleFn {
+  return (
+    isFunction(scale) &&
+    isFunction(scale().copy) &&
+    isFunction(scale().domain) &&
+    isFunction(scale().range)
+  );
+}
+
+function getD3ScaleFromString(scale: ScalePropType): D3ScaleFn {
+  switch (scale) {
+    case "linear":
+      return scaleLinear;
+    case "time":
+      return scaleTime;
+    case "log":
+      return scaleLog;
+    case "sqrt":
+      return scaleSqrt;
+    default:
+      return DEFAULT_SCALE;
+  }
+}
+
+export function useScale(
```

Comment:
Ok, I have a topic to discuss ...
I'm thinking about these new hooks -- `useScale` `useData` `useDomain` and `useRange` ... 

I would expect to use these hooks in my own components.  Like
```
const domain = useDomain();
const scale = useScale();
const range = useRange();
```
And I would kinda expect the values to be coming from the Context.

The way these hooks are written, currently, is to be used by the `VictoryProvider`.  And one thing I notice:  they might not need to be hooks.  These methods don't really use `useState` or `useEffect` or `useContext` ... they primarily just use `useMemo`.

So here's what I'd like to think about -- maybe I'm wrong about my understanding, so please let me know.
If these methods were changed to something like `normalizeScale`, `normalizeDomain`, etc ..., and we removed the `useMemo` out of them, would that make sense?
We would move the `useMemo` into the `VictoryProvider` itself, so that we're memoizing the results of `normalizeScale` etc.

And, then we could export some of these "consumer hooks" like:
```
export function useScale() {
  return useContextSelector(VictoryContext, c => c.scale);
}
```

Let me know if my explanation makes sense!